src/test/java/com/artipie/npm/proxy/NpmProxyTest.java

```diff
@@ -0,0 +1,305 @@
+
+/**
+ * Test NPM Proxy works.
+ *
+ * @since 0.1
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ */
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+@ExtendWith(VertxExtension.class)
+final class NpmProxyTest {
+    /**
+     * Port to listen for Mock Server.
+     */
+    private static final int DEF_PORT = 12_321;
+
+    /**
+     * Last modified date for both package and asset.
+     */
+    private static final String DEF_LAST_MODIFIED = "Tue, 24 Mar 2020 12:15:16 GMT";
+
+    /**
+     * Asset Content-Type.
+     */
+    private static final String DEF_CONTENT_TYPE = "application/octet-stream";
+
+    /**
+     * Assert content.
+     */
+    private static final String DEF_CONTENT = "foobar";
+
+    /**
+     * NPM Proxy instance.
+     */
+    private NpmProxy npm;
+
+    /**
+     * NPM Proxy storage.
+     */
+    private Storage storage;
+
+    @Test
+    public void getsPackage(final VertxTestContext context)
+        throws ParseException, IOException, JSONException {
+        final NpmPackage pkg = this.npm.getPackage("asdas").blockingGet();
+        context.completeNow();
+        MatcherAssert.assertThat("Package is null", pkg != null);
+        MatcherAssert.assertThat(
+            pkg.lastModified(),
+            new IsEqual<>(NpmProxyTest.DEF_LAST_MODIFIED)
+        );
+        final BlockingStorage bstorage = new BlockingStorage(this.storage);
+        final String cached = new String(
+            bstorage.value(new Key.From("asdas/package.json")),
+            StandardCharsets.UTF_8
+        );
+        MatcherAssert.assertThat(
+            "Package content is null",
+            cached != null
+        );
+        JSONAssert.assertEquals(
+            IOUtils.resourceToString(
+                "/cached.json",
+                StandardCharsets.UTF_8
+            ),
+            cached,
+            true
+        );
+        MatcherAssert.assertThat(pkg.data(), new IsEqual<>(cached));
+        final String metadata = new String(
+            bstorage.value(new Key.From("asdas/package.metadata")),
+            StandardCharsets.UTF_8
+        );
+        MatcherAssert.assertThat("Metadata is null", metadata != null);
+        final JSONObject json = new JSONParser(JSONParser.DEFAULT_PERMISSIVE_MODE)
+            .parse(metadata, JSONObject.class);
+        MatcherAssert.assertThat(
+            json.getAsString("last-modified"),
+            new IsEqual<>(NpmProxyTest.DEF_LAST_MODIFIED)
+        );
+    }
+
+    @Test
+    public void cannotFindPackage(final VertxTestContext context) {
+        MatcherAssert.assertThat(
+            "Unexpected package found",
+            this.npm.getPackage("not-found").blockingGet() == null
+        );
+        context.completeNow();
+    }
+
+    @Test
+    public void getsAsset(final VertxTestContext context) throws ParseException {
+        final NpmAsset asset = this.npm.getAsset("asdas/-/asdas-1.0.0.tgz").blockingGet();
+        context.completeNow();
```

Comment:
What is the point of calling this method on the second line of the test?

Reply:
It doesn't matter where this method should be called, but it should be called inside the test method somewhere, because it's expected by vertxextension (https://vertx.io/docs/vertx-junit5/java/). I put it here because of Vertx-related actions are finished here.

Reply:
@brastak , call this method on the last line of the method or remove `context` from the method signature if you don't need it(like in this case)

Reply:
But I need it. When VertxServer is created, specific handler is registered that will be waiting for completion signal. This signal is sent with completeNow call. Without it test will not finish and will be stopped with timeout